src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -97,6 +97,12 @@ internal JsonClassInfo(Type type, JsonSerializerOptions options)
 
                     foreach (PropertyInfo propertyInfo in type.GetProperties(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic))
                     {
+                        // We ignore public indexer
```

Comment:
Nit: all indexers are ignored... it's a little strange to me to just call out "public" here.  I'd suggest just "Ignore indexers".